### ISSUE TITLE
Remove shadow()

### ISSUE THREAD

**Issue by treshugart** [skatejs/skatejs#1496]
I'm proposing we remove `shadow()` and just write that logic directly in `get renderRoot`. It could be useful, but it's trivial to do what it does yourself and would reduce the API surface area.

**Comment by blikblum** [comment]
I have a use case for `shadow`:

Consider a project that uses a Component that does not uses shadow dom:

```javascript
class Component extends withCustomMixin(withLit(withComponent())) {
  get renderRoot() {
     return this
  }
}
```

But in specific cases i want to enable shadow dom, now i would do this:

```javascript
class ShadowComponent extends Component {
  get renderRoot() {
     return shadow(this)
  }
}
```

By removing shadow i would need to be aware to only create the shadow once by caching the returned shadow or returning shadowRoot.

```javascript
class ShadowComponent extends Component {
  get renderRoot() {
     return this.shadowRoot || this.attachShadow({ mode: 'open' })
  }
}
```
Not a big deal but newcomers will need to know a bit about shadow dom behavior

**Comment by treshugart** [comment]
@blikblum couldn't you just do `super.renderRoot` ?

**Comment by sghoweri** [comment]
Hmm. I think I’m going to have to look into this one a bit more carefully.

I know off the top of my head that we do currently use `shadow()` to only use Shadow DOM after checking a couple things at the environment and component level — if you support Shadow DOM natively, if your component doesn’t have a form element as a parent, and if the doesn’t component explicitly opt out of rendering to the Shadow DOM with a `useShadow` prop our components all share.

I’ll have to check out the code for this but as long as we could roll our own version of this (worst case) this doesn’t seem like a deal-breaker.

^ FYI this is why our components use a few built-in slot methods to automatically render slots in our templates if supported / possible and to render the slotted children to the DOM normally if not!

**Comment by blikblum** [comment]
> couldn't you just do super.renderRoot ?

No because super.renderRoot would return this


The API could be simplified to call renderRoot only once in constructor, store and use its value subsequently 

So only two implementations will be possible:

return this
or
return this.attachShadow({ mode: 'open' })

Currently the cache is already done but inside renderRoot 

The `renderRoot` name could be made a method like `getRenderRoot`

**Comment by treshugart** [comment]
> No because super.renderRoot would return this

Ahh yep, if you're returning `this` in the parent, sure.

I don't feel too strongly about this. Even if we put it back, I am thinking the default `renderRoot` behaviour should return a shadow root if available and maybe the element if not for unsupported environments. I've been thinking a lot about SSR and how it could be simpler if you just render to the host, but this is a discussion for another time. Post v6 :)

**Comment by sghoweri** [comment]
@treshugart speaking of SSR, have you seen this?! 
https://github.com/jsdom/jsdom/pull/2347

**Comment by treshugart** [comment]
No, I saw a similar one by someone else that had CE support. That's awesome!

**Comment by treshugart** [comment]
This has already been done.
